frontend/packages/noobaa-storage-plugin/src/components/object-bucket-claim-page/secret.tsx

```diff
@@ -1,41 +1,81 @@
 import * as React from 'react';
 import * as _ from 'lodash';
-import { Base64 } from 'js-base64';
-import { SecretData } from '@console/internal/components/configmap-and-secret-data';
+import { SecretValue } from '@console/internal/components/configmap-and-secret-data';
 import { ConfigMapModel, SecretModel } from '@console/internal/models';
 import { k8sGet, K8sResourceKind } from '@console/internal/module/k8s';
 import { getName, getNamespace } from '@console/shared';
-import { isBound } from '../../utils';
+import { SectionHeading, EmptyBox } from '@console/internal/components/utils';
+import { Button } from '@patternfly/react-core';
+import { EyeSlashIcon, EyeIcon } from '@patternfly/react-icons';
 
 export const GetSecret: React.FC<GetSecretProps> = ({ obj }) => {
-  const secretData = {
-    BUCKET_NAME: Base64.encode(getName(obj)),
-    ACCESS_KEY: Base64.encode('loading..'),
-    SECRET_KEY: Base64.encode('loading..'),
-    ENDPOINT: Base64.encode('http://loading.com'),
-  };
-  if (isBound(obj)) {
+  const [reveal, setReveal] = React.useState(false);
+  const [secretData, setSecretData] = React.useState([]);
+  const bucketName = getName(obj);
+
+  React.useEffect(() => {
     const secret = k8sGet(SecretModel, getName(obj), getNamespace(obj));
     const configMap = k8sGet(ConfigMapModel, getName(obj), getNamespace(obj));
     Promise.all([secret, configMap])
       .then((data) => {
-        secretData.ACCESS_KEY = Base64.encode(_.get(data[0], 'data.AWS_ACCESS_KEY_ID'));
-        secretData.SECRET_KEY = Base64.encode(_.get(data[0], 'data.AWS_SECRET_ACCESS_KEY'));
         const endpoint = `${_.get(data[1], 'data.BUCKET_HOST')}:${_.get(
           data[1],
           'data.BUCKET_PORT',
         )}`;
-        secretData.ENDPOINT = Base64.encode(endpoint);
+        const accessKey = _.get(data[0], 'data.AWS_ACCESS_KEY_ID');
+        const secretKey = _.get(data[0], 'data.AWS_SECRET_ACCESS_KEY');
+        const secretValues = [
+          { name: 'Endpoint', value: endpoint },
+          { name: 'Bucket Name', value: bucketName },
+          { name: 'Access Key', value: accessKey },
+          { name: 'Secret Key', value: secretKey },
+        ];
+        setSecretData(secretValues);
       })
       .catch(() => undefined);
-    return (
-      <div className="co-m-pane__body">
-        <SecretData title="Object Bucket Claim Data" data={secretData} />
-      </div>
-    );
-  }
+    // eslint-disable-next-line
+  }, []);
```

Comment:
We should really have `name` and `namespace` as dependencies

Reply:
If the name of the resource changes then its a completely new resource, a different page would have to be shown. If the user changes the namespace then this resource will not exist at all.

Reply:
I'm not sure how many assumptions I would make about how the component is used since it's an exported component. That might be the case now, but will that always be true? Will this ever be used in another context?

It's really easy to fix. It also guards us against accidentally referring to something other value in the `useEffect` block in the future that might change.

```ts
const name = getName(obj);
const namespace = getNamespace(obj);
React.useEffect(() => {
  // ...
}, [name, namespace];
```

Reply:
The component has very specific fields hardcoded. I do not see much reuse happening. But I have added the dependency list. :)

Reply:
> The component has very specific fields hardcoded. I do not see much reuse happening.

I get that, but this is so easy to do the right way. Just as easy as adding the eslint override :) Then there's never a problem.